### PR TITLE
[release-1.2] 🐛 releaselink: fix goproxy to also return versions for major > 1

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

* Fixes goproxy for `releaselink` to also list versions > v1

Cherry-picks #7709 for releaselink only

cc @fabriziopandini 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: cmd/clusterctl/client/config/cert_manager.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ package config
 // CertManager defines cert-manager configuration.
 type CertManager interface {
 	// URL returns the name of the cert-manager repository.
-	// If empty, "https://github.com/cert-manager/cert-manager/releases/latest/cert-manager.yaml" will be used.
+	// If empty, "https://github.com/cert-manager/cert-manager/releases/{DefaultVersion}/cert-manager.yaml" will be used.
```
Comment (killianmuldoon) [inline cmd/clusterctl/client/config/cert_manager.go:22]: Is this change related to the goproxy changes?

Comment (chrischdi) [reply cmd/clusterctl/client/config/cert_manager.go:22]: Not directly, but was included in the first PR
Due to the first commit which is required in order to apply the second one. 

Original change was at #7192

**Comment by fabriziopandini** [comment]
/hold
I think we should backport only the fix for release link, not the clusterctl part of this PR

**Comment by chrischdi** [comment]
Adjusted to only backport the fix for releaselink.

**Comment by fabriziopandini** [comment]
Great job!
/hold cancel
/lgtm
/approve
